### PR TITLE
[WX-1782] Cloud Billing Dependency + Updates

### PR DESCRIPTION
**Pull request by THWiseman**
### Description
Updated a few dependencies that we need in order to use the Cloud Billing SDK.

#### `CHANGELOG.md`
 - [ ] I updated `CHANGELOG.md` in this PR
 - [x] I assert that this change shouldn't be included in `CHANGELOG.md` because it doesn't impact community users

#### Terra Release Notes
 - [ ] I added a suggested release notes entry in this Jira ticket
 - [x] I assert that this change doesn't need Jira release notes because it doesn't impact Terra users

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -800,9 +790,20 @@ object Dependencies {
   )
 
   private val protobufJavaOverrides = List(
-    "com.google.protobuf" % "protobuf-java" % "3.21.2",
+    "com.google.protobuf" % "protobuf-java" % "3.25.3",
   )
 
+  val servicesDependencies: List[ModuleID] = List(
```
Comment (THWiseman) [inline project/Dependencies.scala:796]: This was moved lower in the file to be below the `googleCloudDependencies` definition. 

Two changes:
- added `"com.google.cloud" % "google-cloud-billing" % "2.47.0"` to the top
- added `++ googleCloudDependencies` to the end

**Review by jgainerdewar — APPROVED** [review]
If CI likes it I like it.

File: project/Dependencies.scala [hunk 2]
```diff
@@ -800,9 +790,20 @@ object Dependencies {
...
+    "com.google.cloud" % "google-cloud-billing" % "2.47.0",
+    "com.google.api" % "gax-grpc" % googleGaxGrpcV,
+    "org.apache.commons" % "commons-csv" % commonsCsvV,
+    "bio.terra" % "bard-client-resttemplate" % bardClientV
+      exclude("org.springframework", "spring-aop")
+      exclude("org.springframework", "spring-jcl"),
+    "org.apache.httpcomponents.client5" % "httpclient5" % apacheHttpClient5V // Needed for rest-template connection pooling
+
+  ) ++ testDatabaseDependencies ++ akkaHttpDependencies ++ mockServerDependencies ++ googleCloudDependencies
```
Comment (aednichols) [inline project/Dependencies.scala:805]: Is it feasible to break up `googleCloudDependencies` and bring only that smaller chunk into services?

Right now its value is `<a bunch of individual stuff>` ++ googleGenomicsV2Alpha1Dependency ++ googleLifeSciencesV2BetaDependency ++ googleBatchv1Dependency` and I'm pretty sure services doesn't need any of the job running stuff.

**Review by aednichols — APPROVED** [review]
LGTM 👍

One optional suggestion that may make life easier in the future.

**Pull request merged**